### PR TITLE
Use interface

### PR DESCRIPTION
**Pull request by juherr**
What do you think if we use interfaces instead of a convention?

### CODE REVIEW

**Comment by sbabcoc** [comment]
That would be more graceful, to be sure. I had considered that, but was wanting this to work with existing classes without modification.

File: src/main/java/org/testng/reporters/IReporterConfig.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.testng.reporters;
```
Comment (sbabcoc) [inline src/main/java/org/testng/reporters/IReporterConfig.java:1]: Not sure that this marker interface provides any value.

Comment (juherr) [reply src/main/java/org/testng/reporters/IReporterConfig.java:1]: Yes, I had the same feeling too.

**Comment by juherr** [comment]
> but was wanting this to work with existing classes without modification.

As I understand, only XmlReporter is using `getConfig`. I can't imagine that another reporter using it, do you?

File: src/main/java/org/testng/reporters/IReporterConfig.java (hunk 1 above)
Comment (sbabcoc) [reply src/main/java/org/testng/reporters/IReporterConfig.java:1]: I've used marker interfaces for classes I enhanced with byte code generators like Byte Buddy, and for classes intended for handling by the ServiceLoader, but this scenario doesn't fit either description. Drop this marker interface, and I think this is ready to merge.

**Comment by sbabcoc** [comment]
Any reporter could have its configuration factored out like this, but `XMLReporter` is currently the only one that uses this strategy. The goal with this PR is to enable reporters that implement this strategy to be instantiated as expected by the `newReporterInstance()` method of **ReporterConfig** objects.

File: src/main/java/org/testng/reporters/IReporterConfig.java (hunk 1 above)
Comment (juherr) [reply src/main/java/org/testng/reporters/IReporterConfig.java:1]: Feel free to remove the interface :)

**Pull request merged**